src/Chapter2.hs

```diff
@@ -25,203 +18,164 @@ In this chapter, you are going to master:
 -}
 
 
 {- |
 =🛡= Lists
-
 __List__ is a crucial data type in Haskell and functional programming
 in general. It represents a collection of elements of the _same type_.
 The list type is written as the type of list element in square
 brackets. For example, a list of integers will have type '[Int]' and a
 list of booleans — '[Bool]'.
-
 [Interesting fact]: String in Haskell is a list of characters ('Char'
 data type in Haskell) and is written as '[Char]'. But Haskell also
 provides the "String" alias to '[Char]'. So, in some places, you will
 see 'String', and in others, you will see '[Char]', but they mean the
 same thing. We will explain better how it works in Chapter Three. For
 now, you only need to know that you can use 'String' and '[Char]'
 interchangeably.
-
 To create a list, you need to put elements in square brackets (the
 same as used for the list type) and separate them by commas. Such
 expressions are called __list literals__. For example, the expression
 "[3, 5, 1]" creates a list of three numbers, where the first list
 element is number 3, and the last element is number 1. Similarly, you
 can create a list of two booleans: [False, True]. A list without
 elements is just [].
-
 You probably noticed that lists could be of any type of
 elements. Often you want to write a function that works with lists of
 any types (but consistent inside one list). This feature is called
 __parametric polymorphism__. It will be explained in more details
 later, but when working with lists, you often will see type signatures
 like:
-
 @
 foo :: [a] -> [b] -> [a]
 @
-
 The above type signature means that this function takes two lists: one
 with elements of some type "a" and another with elements of type "b"
 (the function doesn't care about the specific types) and returns the
 list with elements of the same type as the first list. Such words "a"
 and "b" are called __type variables__.
-
 For comparison, specific types in Haskell start with an uppercase
 letter (Int, Bool, Char, etc.), when type variables begin with a
 lowercase letter (a, b, el, etc.). This is the way to distinguish
 between these types.
-
 The Haskell standard library already provides a lot of functions to
 work with lists. And you will need to operate a lot with standard
 functions in the upcoming exercises. Remember, Hoogle is your friend!
 -}
 
 {- |
 =⚔️= Task 1
-
 Explore lists by checking types of various list expressions and
 functions in GHCi and insert the corresponding resulting output below:
-
 List of booleans:
 >>> :t [True, False]
-
-
+[True, False] :: [Bool]
 String is a list of characters:
 >>> :t "some string"
-
-
+"some string" :: [Char]
 Empty list:
 >>> :t []
-
-
+[] :: [a]
 Append two lists:
 >>> :t (++)
-
-
+(++) :: [a] -> [a] -> [a]
 Prepend an element at the beginning of a list:
 >>> :t (:)
-
-
+(:) :: a -> [a] -> [a]
 Reverse a list:
 >>> :t reverse
-
-
+reverse :: [a] -> [a]
 Take first N elements of a list:
 >>> :t take
-
-
```

Comment:
I see that empty lines were removed here. Unfortunately, because of this, Doctests on CI will fail :disappointed: They need an empty line after the output and before the next textual block